include/robot_interfaces/robot_backend.hpp

```diff
@@ -47,18 +47,27 @@ class RobotBackend
      * @param robot_data  Data is send to/retrieved from here.
      * @param max_action_duration_s  See MonitoredRobotDriver.
      * @param max_inter_action_duration_s  See MonitoredRobotDriver.
+     * @param real_time_mode  Enable/disable real-time mode.  In real-time mode,
+     *     the backend will repeat previous actions if the new one is not
+     *     provided in time or fail with an error if the allowed number of
+     *     repetitions is exceeded.  In non-real-time mode, it will simply block
+     *     and wait until the action is provided.
      */
-    RobotBackend(
-        std::shared_ptr<RobotDriver<Action, Observation>> robot_driver,
-        std::shared_ptr<RobotData<Action, Observation>> robot_data,
-        const double max_action_duration_s,
-        const double max_inter_action_duration_s)
+    RobotBackend(std::shared_ptr<RobotDriver<Action, Observation>> robot_driver,
+                 std::shared_ptr<RobotData<Action, Observation>> robot_data,
+                 const double max_action_duration_s,
+                 const double max_inter_action_duration_s,
+                 const bool real_time_mode = true)
```

Comment:
i am worried that this is becoming slightly obscure, i.e. it is not obvious at first sight how these parameters interact. 

would it make sense to e.g. pass it directly a monitored driver, such that it does not need to construct it internally, and hence it would not need the arguments max_action_duration_s, max_inter_action_duration_s? this would encapsulate parameters more. also, i think it would be good to have max_action_repetitions as an argument here as well, then everything is together.

i will approve anyway, but maybe this would be nice for the future, because for somebody who is not used to it, i think this is a bit confusing.

Reply:
in fact, if we pass it directly a monitored/unmonitored driver, and it does not need to do construction internally, then this safety check is abstracted away. for a simulator for instance, you don't need to monitor the driver, you can directly pass it the unmonitored driver. so i think this would make things much easier to understand, and it makes sense particularly also for the simulation driver.

Reply:
Sounds good, I will do this in a separate PR.